MaterialLib/PorousMedium/UnsaturatedProperty/RelativePermeability/CreateRelativePermeabilityModel.cpp

```diff
@@ -0,0 +1,227 @@
+*/
+std::unique_ptr<RelativePermeability> createNonWettingPhaseBrookCoreyOilGas(
+    BaseLib::ConfigTree const& config)
+{
+    //! \ogs_file_param{material_property__porous_medium__porous_medium__relative_permeability__type}
+    config.checkConfigParameter("type", "NonWettingPhaseBrookCoreyOilGas");
+
+    //! \ogs_file_param{material_property__porous_medium__porous_medium__relative_permeability__NonWettingPhaseBrookCoreyOilGas__sr}
+    const double Sr = config.getConfigParameter<double>("sr");
+
+    //! \ogs_file_param{material_property__porous_medium__porous_medium__relative_permeability__NonWettingPhaseBrookCoreyOilGas__smax}
+    const double Smax = config.getConfigParameter<double>("smax");
+
+    //! \ogs_file_param{material_property__porous_medium__porous_medium__relative_permeability__NonWettingPhaseBrookCoreyOilGas__m}
+    const double m = config.getConfigParameter<double>("m");
+    if (m < 1.0)  // m >= 1
+    {
+        OGS_FATAL(
+            "The exponent parameter of NonWettingPhaseBrookCoreyOilGas\n"
+            "relative permeability model, m, must not be smaller than 1");
+    }
+    //! \ogs_file_param{material_property__porous_medium__porous_medium__relative_permeability__NonWettingPhaseBrookCoreyOilGas__m}
+    const double krel_max = config.getConfigParameter<double>("krel_min");
+
+    return std::unique_ptr<RelativePermeability>(
+        new NonWettingPhaseBrookCoreyOilGas(Sr, Smax, m, krel_max));
+}
+
+std::unique_ptr<RelativePermeability> createRelativePermeabilityModel(
+    BaseLib::ConfigTree const& config)
+{
+    //! \ogs_file_param{material_property__porous_medium__porous_medium__relative_permeability__type}
+    auto const type = config.peekConfigParameter<std::string>("type");
+
+    if (type == "WettingPhaseVanGenuchten")
+    {
+        return createWettingPhaseVanGenuchten(config);
+    }
+    else if (type == "NonWettingPhaseVanGenuchten")
+    {
+        return createNonWettingPhaseVanGenuchten(config);
+    }
+    else if (type == "WettingPhaseBrookCoreyOilGas")
+    {
+        return createWettingPhaseBrookCoreyOilGas(config);
+    }
+    else if (type == "NonWettingPhaseBrookCoreyOilGas")
+    {
+        return createNonWettingPhaseBrookCoreyOilGas(config);
+    }
+    else if (type == "Curve")
+    {
+        config.checkConfigParameter("type", "Curve");
+
+        std::vector<double> variables, values;
+        //! \ogs_file_param{material_property__porous_medium__porous_medium__relative_permeability__Curve__curve}
+        auto const& curve_config = config.getConfigSubtree("curve");
+        for (
+            auto const& data_string :
+            //! \ogs_file_param{material_property__porous_medium__porous_medium__relative_permeability__Curve_curve__data}
+            curve_config.getConfigParameterList<std::string>("data"))
+        {
+            std::stringstream ss(data_string);
+            double var, val;
+            ss >> var >> val;
```

Comment:
getConfigParameterList<std::vector<double>> does not work. Changed syntax to
```
             <curve>
                   <point><data> 0. 0.9   </data></point>
                   <point><data> 0.4 0.5  </data></point>
                   <point><data> 0.9 0.01 </data></point>
              </curve>
```
and read it by
```
        for (  auto const& point_config :  curve_config.getConfigSubtreeList("point"))
        {
            const auto& point =
                               point_config.getConfigParameter<std::vector<double>>("data");
            assert(point.size() == 2);
           .... 
       }
```

I think, reading curve data in pair is easy for manually checking.